src/i18_bluesky/plans/align_lookup_2.py

```diff
@@ -0,0 +1,127 @@
+from dataclasses import dataclass
+from typing import List, Optional
+
+import bluesky.plan_stubs as bps
+import numpy as np
+import sympy as sp
+from bluesky.devices.monochromator import DCM
+from dodal.common import MsgGenerator, inject
+
+monochromator = inject("monochromator")
+undulator = inject("undulator")
+pinhole = inject("pinhole")
+KBMirror = inject("kb_mirror")
+
+
+def calculate_derivative_maxima(data: np.ndarray) -> List[float]:
+    x = sp.Symbol("x")
+    y = sp.interpolating_spline(3, sp.lambdify(x, data))
+    derivative = y.diff(x)
+    critical_points = sp.solve(derivative, x)
+    return [float(cp) for cp in critical_points if cp.is_real]
+
+
+@dataclass
+class BeamlineAlignmentParams:
+    target_energy: Optional[float] = 12.0
+    tolerance: Optional[float] = 0.1
+
+
+def calibrate_monochromator(monochromator: DCM) -> MsgGenerator:
+    yield from bps.mv(monochromator, "calibrate_dcm")
+    # todo this measure foil makes no sense at the moment
+    yield from bps.mv(monochromator, "measure_foil", "Fe")
+    yield from bps.mv(monochromator, "measure_foil", "Mg")
+
+
+def adjust_bragg_offset_using_absorption_spectrum(
+    monochromator,
+    absorption_spectrum: np.ndarray,
+) -> MsgGenerator:
+    energy_positions = calculate_derivative_maxima(absorption_spectrum)
+    for position in energy_positions:
+        yield from bps.mv(monochromator.bragg_in_degrees, position)
```

Comment:
just move to the 1 position with the highest energy that was found from calculate_derivative_maxima, don't sweep through all positions